thoth/package_extract/core.py

```diff
@@ -47,20 +48,30 @@ def extract_buildlog(input_text: str) -> typing.List[dict]:
 def extract_image(image_name: str, timeout: int = None, *, registry_credentials: str = None,
                   tls_verify: bool=True) -> dict:
     """Extract dependencies from an image."""
-    image_name = quote(image_name)
-    with tempfile.TemporaryDirectory() as dir_path:
+    #Setting up the prometheus registry and the Gauge metric
+    prometheus_registry = CollectorRegistry()
+    metric_analyzer_job = Gauge('package_extract_time', 'Runtime of package extract job', registry=prometheus_registry)
+    #Begins a timer to record the running time of the job
+    with metric_analyzer_job.time(), tempfile.TemporaryDirectory() as dir_path:
+        image_name = quote(image_name)
         download_image(
             image_name,
             dir_path,
             timeout=timeout or None,
             registry_credentials=registry_credentials or None,
             tls_verify=tls_verify
         )
-
         rootfs_path = os.path.join(dir_path, 'rootfs')
         layers = construct_rootfs(dir_path, rootfs_path)
 
         result = run_analyzers(rootfs_path)
         result['layers'] = layers
-
-        return result
+        
+   push_gateway = os.getenv('PROMETHEUS_PUSH_GATEWAY', 'pushgateway:9091')
+   if push_gateway:
```

Comment:
Won't `push_gateway` be always true, because you default it with `os.getenv` above?

Reply:
If the `PROMETHEUS_PUSH_GATEWAY` will be set to empty string, it will be evaluated as false.